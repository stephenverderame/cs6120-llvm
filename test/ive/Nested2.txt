30
80
81
150
151
152
240
241
242
243
350
351
352
353
354
480
481
482
483
484
485
630
631
632
633
634
635
636
800
801
802
803
804
805
806
807
990
991
992
993
994
995
996
997
998
