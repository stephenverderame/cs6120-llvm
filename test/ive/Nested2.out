Loop: 
7:                                                ; preds = %14, %4
  %.0 = phi i32 [ 0, %4 ], [ %15, %14 ]
  %8 = icmp slt i32 %.0, %.01
  br i1 %8, label %10, label %9

Basic IV:   %.0 = phi i32 [ 0, %4 ], [ %15, %14 ]
Basic IV:   %15 = add nsw i32 %.0, 1
Loop: 
1:                                                ; preds = %17, %0
  %.01 = phi i32 [ 0, %0 ], [ %18, %17 ]
  %2 = icmp slt i32 %.01, 10
  br i1 %2, label %4, label %3

Basic IV:   %.01 = phi i32 [ 0, %0 ], [ %18, %17 ]
Basic IV:   %18 = add nsw i32 %.01, 1
Derived IV: %5 = %.01 * 10
Derived IV: %6 = %.01 * 10 + 20
